sdk/storage/azfile/sas/account_test.go

```diff
@@ -0,0 +1,92 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package sas
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestAccountPermissions_String(t *testing.T) {
+	testdata := []struct {
+		input    AccountPermissions
+		expected string
+	}{
+		{input: AccountPermissions{Read: true}, expected: "r"},
+		{input: AccountPermissions{Write: true}, expected: "w"},
+		{input: AccountPermissions{Delete: true}, expected: "d"},
+		{input: AccountPermissions{List: true}, expected: "l"},
+		{input: AccountPermissions{Create: true}, expected: "c"},
+		{input: AccountPermissions{
+			Read:   true,
+			Write:  true,
+			Delete: true,
+			List:   true,
+			Create: true,
+		}, expected: "rwdlc"},
+	}
+	for _, c := range testdata {
+		require.Equal(t, c.expected, c.input.String())
+	}
+}
+
+func TestAccountPermissions_Parse(t *testing.T) {
+	testdata := []struct {
+		input    string
+		expected AccountPermissions
+	}{
+		{expected: AccountPermissions{Read: true}, input: "r"},
+		{expected: AccountPermissions{Write: true}, input: "w"},
+		{expected: AccountPermissions{Delete: true}, input: "d"},
+		{expected: AccountPermissions{List: true}, input: "l"},
+		{expected: AccountPermissions{Create: true}, input: "c"},
+		{expected: AccountPermissions{
+			Read:   true,
+			Write:  true,
+			Delete: true,
+			List:   true,
+			Create: true,
+		}, input: "rwdlc"},
+		{expected: AccountPermissions{
+			Read:   true,
+			Write:  true,
+			Delete: true,
+			List:   true,
+			Create: true,
+		}, input: "rwldc"},
```

Comment:
randomize the order here to test that it works

Reply:
Done